sdk/rust/oak/src/io/receiver.rs

```diff
@@ -153,6 +161,11 @@ impl<T: Decodable> ReceiverExt<T> for Receiver<T> {
         }
     }
 
+    fn receive_with_downgrade(&self) -> Result<T, OakError> {
+        self.wait()?;
```

Comment:
Should there not be a `wait_with_downgrade`? Knowing that a message has become available leaks information as well, so a node should not be able to wait on a channel from which it cannot read.

Reply:
I implemented it in https://github.com/project-oak/oak/pull/1839 (will add after the merge)
Will add a TODO in this PR